capsules/extra/src/ethernet_tap.rs

```diff
@@ -0,0 +1,657 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Leon Schuermann <leon@is.currently.online> 2025.
+// Copyright Tock Contributors 2025.
+
+//! # Userspace application TAP-like network driver
+//!
+//! This capsule provides a userspace syscall driver designed to expose a raw
+//! IEEE 802.3 (Ethernet) compatible network interface (resembling a
+//! Linux/FreeBSD TAP virtual network device).
+//!
+//! The driver multiplexes the underlying network interfaces across all
+//! applications. Applications can allow a receive buffer to the kernel, and all
+//! incoming frames will be placed into all applications' receive buffers. Every
+//! application may transmit frames, which are sent on the underlying Ethernet
+//! link.
+//!
+//! Incoming frames are sent to all processes (effectively mimicking the
+//! behavior of a MacVTap device in Virtual Ethernet Port Aggregator (VEPA)
+//! mode). This means that all processes can communicate over the Ethernet link,
+//! but processes will generally not able to talk to each other, _unless_ the
+//! connected far-end Ethernet device supports hairpin mode and is able to send
+//! frames back on the link they came from. If processes should be able to talk
+//! to each other, we can instantiate a hairpinning bridge in the Tock kernel,
+//! or use a regular bridge and provide each process its own, dedicated tap
+//! driver.
+//!
+//! ## Interface description
+//!
+//! This driver can both deliver incoming IEEE 802.1 Ethernet frames from the
```

Comment:
Did you mean 802.3 here?

```suggestion
//! This driver can both deliver incoming IEEE 802.3 Ethernet frames from the
```